apps/sms/views/conversation/js/compose.js

```diff
@@ -4,10 +4,7 @@
 /*global Settings, Utils, Attachment, MozActivity, SMIL,
```

Comment:
Something may not work right now (definitely!) and naming maybe confusing, so let me describe the idea I have in mind and check first if it makes sense for you at all :)

I consider ```Compose``` as all-sufficient component that can be used by different views. It has knowledge only about its nested components (input, subject, send and attach buttons). It also has some _smart_  rules that define its behavior (e.g. if subject is not empty - message should be ```MMS```, if size of message is too big - attach button should be disabled).

But since this component can be used in different _contexts_ (for now Conversation and NewMessage views) it should allow _context_ to alter its own _smart_ rules with custom behavior (e.g NewMessage view would like to disable send button if there is no recipients in another all-sufficient and isolated ```Recipients``` component or Conversation view would like to force ```MMS``` type if it's group or email conversation or maybe we'll have switch somewhere outside of ```Compose``` to always force ```MMS```), no matter what default _smart_ rules dictate.

So the usual way to make component all-sufficient is to remove dependencies to another components/context and use dependency injection to alter its behavior. Essentially we can have either "constructor injection" or "method injection" ("property injection" sounds quite clumsy for me in this context :)).

Let me start from the way I chose for this PR - ```method injection```. I found the idea of ```locking``` appealing (like simplified version of altering behaviour via simply forcing something) - when component behavior can be forced/locked from outside. I introduced ```Compose.setupLock({send, edit, type})``` method (not sure about name though) that allows context to forbid sending, editing (....though only if message size is too big) and message type depending one something that context aware of, but ```Compose``` doesn't know. Initially I thought that only settings (things that also alter behavior, but aren't changed during app run time) can be passed through constructor (email support or mms max size).

Regarding ```constructor injection``` - I was thinking we way pass some "delegates" functions like ```ensureType```, ```canSend``` or ```canEdit``` that our views will pass into ```Compose``` constructor. Or something similar, but I haven't thought about possible caveats here yet. 

As far as I understood you wanted to pass entire view (that implements some predefined interface to access recipients in a unified way) as parameter to ```Compose``` constructor. I'm not really sure about the following things in this solution: ```Compose``` will know what ```recipients``` is and it will have access to entire view method/property set. But maybe I'm missing something.

I don't have too strong opinion on any solution yet, so would be happy to hear your thoughts/ideas/objections!